### ISSUE TITLE
Doesn't build on non-amd64

### ISSUE THREAD

**Issue by calmh** [twmb/murmur3#6]
```
jb@kvin:~/dev/tmp/murmur3 % GOOS=linux GOARCH=386 go build
# github.com/twmb/murmur3
./murmur128_gen.go:16:51: undefined: unsafe
./murmur128_gen.go:25:43: undefined: unsafe
jb@kvin:~/dev/tmp/murmur3 % git describe
v1.1.2
jb@kvin:~/dev/tmp/murmur3 %
```

The generic implementation does an unsafe conversion from []byte to string.

Frankly I'm not sure the `...String` functions have much use, and the one implementation could instead use byte slices and not need to do this magic. Or the implementation could just be switched to take []byte and the ...String methods could do a non-unsafe conversion or do the "normal" unsafe dance with reflect.SliceHeader and such.

**Comment by twmb** [comment]
I'd rather avoid the garbage, which was part of the purpose of my fork of this repo. Although, I acknowledge that avoiding the garbage could be punted outside of this package (users could do the unsafe dance), but there isn't much harm leaving it inside this package either.

Good catch on the build failure, though; how I missed that is beyond me.

**Comment by calmh** [comment]
I don't mind fixing it, either by removing the string-typed functions (API breakage) or making them the obvious-but-slow path and the byte slice version the fast path. My uses are only indirect, but I don't *think* any of them use the string versions. :)

**Comment by twmb** [comment]
What's the benefit of having `[]byte` be the fast path and `string` the slow path?

**Comment by calmh** [comment]
Only that I expect []byte to be the more common use case, but it's not based on anything scientific.

**Comment by twmb** [comment]
I see, so the main problem here is the extra time in the function call overhead? I thought you were trying to get at the theoretically unsound usage of the unsafe conversion (I'm aware of https://github.com/golang/go/issues/19367 and monitoring that though).

**Comment by calmh** [comment]
Yeah the unsafe usage is unsound so it has to go. Doing it with unsafe+the reflect types should be sound but even I am not sure about all the pitfalls with what needs to happen in a single expression and whatnot so I don't trust myself to get it right. The thing I could do is the trivial but slow solution and then we can pick either string or []byte to be fast, or duplicate the implementation...

**Comment by twmb** [comment]
Has to go is a bit of a stretch. There's quite a lot of Go code in existence using that unsafe trick. Although no doc explicitly says it is safe, and they do discourage it, I believe it will break so much code if they ever change things that they're stuck.

However, benchmarking the proper reflect way of doing things shows that it's a 0.23ns conversion, so I can easily swap it out. Older versions of this code used a proper reflect conversion, and that's because older versions of this code actually hashed on a slice instead of a string.

Hashing on a string is beneficial because it removes one instruction from the hot loop: https://github.com/twmb/murmur3/commit/5e319d659499bfc0c17dea37a2a7c00046a79606

I'll re-introduce the absolute correctness because the performance hit is so small, but keeping the hashing on strings is beneficial.

PR incoming soon (just ran benchmarks against the spaolacci code again).

**Comment by twmb** [comment]
All issues in this issue have been addressed with release `v1.1.3`
